CIP-SUBLEDGER/README.md

```diff
@@ -0,0 +1,144 @@
+
+# Specification
+## Objectives
+This CIP encompasses:
+- An implementation of a smart contract system (subledger)
+- The particular build & deployment of that smart contract
+- A guide for usage and integration by SDKs, wallets, and the ecosystem
+
+## Proposed Implementation
+
+### Identifying Seizable Assets
+- Seizable assets are identified by a prefix on the asset name.
+  - To prevent 'accidental' membership in the seizure set, a long prefix and a checksum are recommended.
+
+### Authorization of Seizure Capabilities
+- Seizure capabilities are authorized via multi-validators on the asset policy (withdrawals).
+
+### The RWA Script
+We have a universal script, henceforth the "RWA Script", which maintains the capabilities of EUTXO by specifying in the datum either the signing key or the withdrawal that must be invoked in lieu of a spending validator.
+
+#### Datum Structure
+```rs
+type Datum {
+    // verification_key: check extra_signatories
+    // script_hash: check for withdrawal of 0
+    owner: Credential,
+    datum: Data
+    // usually empty, allows sane handling of seizure
+    seized: Map<ByteArray, Int>
+}
+```
+
+### Constraints on Spending
+- Any assets in the UTxO must **remain** at a new UTxO in the RWA Script. They are only allowed to change the "owner" field; the `output.address.payment_credential` cannot change.
+- This is enforced by folding over the input value set (at the script), the mint value, and the output values (at the script):
+  - `input_values + mint_value - output_values`
+  - Remove negatives
+  - Any positives must not match the "prefix" condition that identifies seizable assets
+- It is impossible to define clawback/seizure for existing assets, particularly for ADA. Hence we can ignore fee/withdrawals.
+
+This guarantees that you can always utilize the policy-level functionality on the UTxOs.
+
+
+### Policy-Level Functionality
+
+#### Additional Constraint (Redeemer Endpoint)
+- Value may be withdrawn from a UTxO.
+- Must create a continuing output with the un-seized assets.
+- Must insert the seized assets into the 'seized' map.
+- For that value to be withdrawn, we must invoke the same minting policy those assets are under, except under the "withdrawal" endpoint. Typically, a withdrawal of 0 is used.
+
+#### Redeemer Structure
+
+```rs
+type Redeemer {
+    withdrawals: List<AdminSeizures>,
+    extra: Data
+}
+
+type AdminSeizures {
+    input_oref: OutputReference,
+    assets: List<ByteArray>
+}
+```
+- Where assets represent the list of asset names being withdrawn.
+- The extra field is for arbitrary extra redeemer data
+- Extra field example: could be used for a signed payload if it's an admin endpoint using a key.
+
+## Development
+The primary improvement for developers is the ability to specify all these "capabilities to freeze/unfreeze" externally to the user-level experience of transferring the UTxOs.
```

Comment:
This first of 3 items is apparently Motivation for the CIP and should be moved to that section...